miro_preprocessor/miro_copy_s3_asset/miro_copy_s3_master_asset/src/miro_copy_s3_master_asset.py

```diff
@@ -0,0 +1,27 @@
+from functools import partial
+import json
+import os
+
+import boto3
+
+from wellcome_lambda_utils.miro_utils import MiroImage
+from wellcome_lambda_utils.s3_utils import S3_Identifier
+from wellcome_lambda_utils import s3_utils
+
+
+def main(event, _):
+    print(f"Received event:\n{event}")
+    s3_client = boto3.client("s3")
+    source_bucket_name = os.environ["S3_SOURCE_BUCKET"]
+    destination_bucket_name = os.environ["S3_DESTINATION_BUCKET"]
+    destination_prefix = os.environ["S3_DESTINATION_PREFIX"]
+
+    image_info = json.loads(event['Records'][0]['Sns']['Message'])
+    miro_image = MiroImage(image_info)
+    key = f"Wellcome_Images_Archive/{miro_image.collection} Images/{miro_image.image_path}.jp2"
+    print(key)
+    destination_key = f"{destination_prefix}/{miro_image.image_path}.jp2"
+    source_identifier = S3_Identifier(source_bucket_name, key)
+    destination_identifier = S3_Identifier(destination_bucket_name, destination_key)
+    s3_utils.exec_if_key_exists(s3_client, source_identifier=source_identifier,
+                                function=partial(s3_utils.copy_asset_if_not_exists, destination_identifier=destination_identifier))
```

Comment:
This looks similar to `miro_copy_s3_derivative_asset.py`, but I thought the idea was that we’d have two Lambdas feeding from the same source tree?

Reply:
I'm not sure I follow. Initially, the idea was to instantiate two lambdas from the same code, but it quickly turned out to be tricky to make it generic enough as there are significant differences in where the source images are for master and derivative assets. 
So, what I've done in this PR is having a different source code for copying master and derivatives respectively. Both of those can then be instantiated as many times as we need. For example, the copy master asset will have to be instantiated three times (digital library, tandem_vault, cold_store)